.github/workflows/integration.yml

```diff
@@ -0,0 +1,126 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Integration Tests
+
+on:
+  push:
+    branches:
+    - 'main'
+  pull_request:
+
+jobs:
+  collatz:
+    name: Build collatz binary
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        name: Checkout collatz example
+        with:
+          repository: ServiceWeaver/weaver
+          ref: 'v0.15.0'
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.20'
+          cache: true
+      - name: Build collatz example
+        working-directory: ./examples/collatz
+        run: |
+          # Disable CGO to ensure the binary is statically linked
+          CGO_ENABLED=0 \
+          go build -o collatz
+      - uses: actions/upload-artifact@v3
+        with:
+          name: collatz-binary
+          path: ./examples/collatz/collatz
+  deploy:
+    name: Deploy and run tests
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    needs: collatz
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.20'
+          cache: true
+      - uses: azure/setup-kubectl@v3
+      - uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Install weaver-kube
+        run: |
+          go install ./cmd/weaver-kube
+      - uses: actions/download-artifact@v3
+        with:
+          name: collatz-binary
+          path: ./examples/collatz
+      - uses: ASzc/change-string-case-action@v5
+        id: repository
+        with:
+          string: ${{ github.repository }}
+      - name: Generate weaver.toml
+        uses: DamianReeves/write-file-action@master
+        with:
+          path: ./examples/collatz/weaver.toml
+          write-mode: overwrite
+          contents: |
+            [serviceweaver]
+            binary = "./collatz"
+            [kube]
+            image = "ghcr.io/${{ steps.repository.outputs.lowercase }}/example-collatz"
+            listeners.collatz = {public = true}
+      - name: Build and generate kube manifests
+        working-directory: ./examples/collatz
+        run: |
+          # Ensure the binary is executable
+          chmod +x ./collatz
+          # Build the docker image and push
+          weaver-kube deploy weaver.toml
```

Comment:
You mentioned this step is failing because it is not permitted to upload a container to ghcr.io/ServiceWeaver/weaver-kube/example-collatz. What do we have to do to make that work?

Reply:
Hey, It just because I don't have write permission on this repo. So, my PR also don't have permission to create a new package for storing docker image.

Reply:
Ah gotcha. Thanks for explaining!